src/Microsoft.Health.Dicom.Core/Features/CustomTag/DeleteCustomTagService.cs

```diff
@@ -0,0 +1,78 @@
+    {
+        private readonly ICustomTagStore _customTagStore;
+        private readonly IDicomTagParser _dicomTagParser;
+        private readonly ILogger<DeleteCustomTagService> _logger;
+
+        /// <summary>
+        /// Mapping from CustomTagVR to IndexType
+        /// </summary>
+        private static readonly IReadOnlyDictionary<string, CustomTagIndexType> CustomTagVRAndIndexTypeMapping = new Dictionary<string, CustomTagIndexType>()
+            {
+                { DicomVR.AE.Code, CustomTagIndexType.StringIndex },
+                { DicomVR.AS.Code, CustomTagIndexType.StringIndex },
+                { DicomVR.AT.Code, CustomTagIndexType.LongIndex },
+                { DicomVR.CS.Code, CustomTagIndexType.StringIndex },
+                { DicomVR.DA.Code, CustomTagIndexType.DateTimeIndex },
+                { DicomVR.DS.Code, CustomTagIndexType.StringIndex },
+                { DicomVR.DT.Code, CustomTagIndexType.DateTimeIndex },
+                { DicomVR.FL.Code, CustomTagIndexType.DoubleIndex },
+                { DicomVR.FD.Code, CustomTagIndexType.DoubleIndex },
+                { DicomVR.IS.Code, CustomTagIndexType.StringIndex },
+                { DicomVR.LO.Code, CustomTagIndexType.StringIndex },
+                { DicomVR.PN.Code, CustomTagIndexType.StringIndex },
+                { DicomVR.SH.Code, CustomTagIndexType.StringIndex },
+                { DicomVR.SL.Code, CustomTagIndexType.LongIndex },
+                { DicomVR.SS.Code, CustomTagIndexType.LongIndex },
+                { DicomVR.TM.Code, CustomTagIndexType.DateTimeIndex },
+                { DicomVR.UI.Code, CustomTagIndexType.StringIndex },
+                { DicomVR.UL.Code, CustomTagIndexType.LongIndex },
+                { DicomVR.US.Code, CustomTagIndexType.LongIndex },
+            };
+
+        public DeleteCustomTagService(ICustomTagStore customTagStore, IDicomTagParser dicomTagParser, ILogger<DeleteCustomTagService> logger)
+        {
+            EnsureArg.IsNotNull(customTagStore, nameof(customTagStore));
+            EnsureArg.IsNotNull(dicomTagParser, nameof(dicomTagParser));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _customTagStore = customTagStore;
+            _dicomTagParser = dicomTagParser;
+            _logger = logger;
+        }
+
+        public async Task DeleteCustomTagAsync(string tagPath, CancellationToken cancellationToken)
+        {
+            DicomTag[] tags;
+            if (!_dicomTagParser.TryParse(tagPath, out tags, supportMultiple: false))
+            {
+                throw new InvalidCustomTagPathException(
+                    string.Format(CultureInfo.InvariantCulture, DicomCoreResource.InvalidCustomTag, tagPath ?? string.Empty));
+            }
+
+            string normalizedPath = tags[0].GetPath();
+
+            CustomTagEntry customTagEntry = await _customTagStore.GetCustomTagAsync(normalizedPath, cancellationToken);
```

Comment:
Does this throw if the tag is not available? If not we should handle it here.

Reply:
the GetCustomTagAsync should throw exception if not exist.